attractions/form-field/form-field.svelte

```diff
@@ -3,15 +3,48 @@
 
   let _class = null;
   export { _class as class };
+  /**
+   * A class string to add to the `<label>` containing the form field name.
+   * @type {string}
+   */
   export let nameClass = null;
+  /**
+   * A class string to add to the help text of the form field.
+   * @type {string}
+   */
   export let helpClass = null;
 
+  /**
+   * The name of the form field. Displayed prominently next to the actual field.
+   * @type {string}
+   */
   export let name = null;
+  /**
+   * The subtitle text under the name providing extra guidance.
+   * @type {string}
+   */
   export let help = null;
+  /**
+   * The ID to add the the `for` attribute of the `<label>` element containing the `name`.
+   * Useful if you pass an ID to the actual field inside and want to connect it with the label.
+   * @type {string}
+   */
   export let id = null;
 
+  /**
+   * Marks the form field as required, adding an asterisk to the name and text under the field indicating that the field is mandatory.
+   * @type {boolean}
+   */
   export let required = false;
+  /**
+   * Marks the form field as optional, explicitly stating the optionality of the field in the text under the field.
+   * @type {boolean}
+   */
   export let optional = false;
+  /**
+   * An array of error messages to display under the field. Falsy elements are not shown.
+   * @type {string[]}
```

Comment:
It is not exactly `false`, but any falsey value. This includes `false`, `null`, `undefined`, `0`, `""`, and `NaN`. There is no type for that in typescript, and it feels too ugly to write out entirely. Either we create an alias `type Falsy = false | null | undefined | 0 | ""` (note that `NaN` cannot be used as a type), or we limit it to just exactly `false`, in which case we need to update a lot of the documentation
Note that using `null` will now throw an error unless `strictNullChecks` is on, which would actually break most of our types.

Reply:
I know, but the main usecase for this thing is writing `errors = [condition && 'error', another && 'other']`, so it's `false` that we need to allow mostly.

Reply:
Why would `strictNullChecks` break our types?

Reply:
Because `let x: string = null` is not valid with `strictNullChecks`, which we have almost everywhere.
```js
/** @type {string} */
export let value = null;
```
is basically doing just that.
Although the issue may not be that severe if we don't expect the users to pass in `null` at all and we just use it as a default value.

Reply:
Well, that says something about our typings. Perhaps we should add `null` in many places since it is technically a valid value for a lot of props

Reply:
Maybe, but it also doesn't always make much sense to pass `null` for something when you can just leave it. The only problem might be when a user binds a variable whose type is `T|null` and our prop accepts just `T` (with `strictNullChecks` of course), but I don't know how often could this be. I am also not sure if Svelte itself will be strict somehow. I think it is best to either test it or wait until someone opens an issue

Reply:
I'll mark this as resolved (as the main comment has been addressed), and we can discuss the `strictNullChecks` issue if a related issue arises.